hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -559,18 +568,28 @@ public OutputStream createFile(final Path path,
             isNamespaceEnabled ? getOctalNotation(permission) : null,
             isNamespaceEnabled ? getOctalNotation(umask) : null,
             isAppendBlob,
+            metadata,
             tracingContext
         );
 
       } else {
-        op = client.createPath(relativePath, true,
-            overwrite,
-            isNamespaceEnabled ? getOctalNotation(permission) : null,
-            isNamespaceEnabled ? getOctalNotation(umask) : null,
-            isAppendBlob,
-            null,
-            tracingContext);
-
+        if (getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
This should be Blob && !isNamespace, as the changes are not scoped for HNS + blob

Reply:
PrefixMode would be BLOB only for FNS accounts. This variable is set in the initialize method of AzureBlobFileSystem. So we don't explicit check for isNamespaceEnabled.